03-first-task.Rmd

```diff
@@ -417,4 +418,60 @@ task BwaMem {
     disks: "local-disk 100 SSD"
   }
 }
-  ```
\ No newline at end of file
+```
+
+## Putting the workflow together
+
+A workflow is needed to run the `BwaMem` task we just built. The workflow's input variables are defined by the workflow JSON metadata, and are then passed on as inputs in our `BwaMem` call. When the `BwaMem` call is complete, the workflow's output File variable is defined based on the task's output. Lastly, we have a parameter_meta component in our workflow that describes each workflow input variable as documentation.
+
+```         
+version 1.0
+
+workflow minidata_test_alignment {
+  input {
+    # Sample info
+    File sampleFastq
+    # Reference Genome information
+    File ref_fasta
+    File ref_fasta_index
+    File ref_dict
+    File ref_amb
+    File ref_ann
+    File ref_bwt
+    File ref_pac
+    File ref_sa
```

Comment:
We may want to consider adding `Int? bwa_mem_threads` here.

You can still edit task-level inputs that are not workflow-level inputs, but the mechanism for doing so can vary slightly depending on the executor. To keep things simple, I think we should add it here for now. As the workflow gets longer, we can discuss when it's appropriate to pare down the number of workflow-level inputs.